execution-engine/src/wasi/common.rs

```diff
@@ -0,0 +1,1519 @@
+        memory_ref.write_string_list(&buffer, buf_address, address_for_string_ptrs)
+    }
+
+    /// The implementation of the WASI `args_sizes_get` function. It requires an extra `memory_ref` to
+    /// interact with the execution engine.                  
+    pub(crate) fn args_sizes_get(
+        &self,
+        memory_ref: &mut impl MemoryHandler,
+        address_for_counts: u32,
+        address_for_buffer_size: u32,
+    ) -> ErrNo {
+        info!("args_sizes_get is called");
+        let environc = self.program_arguments.len() as u32;
+        let environ_buf_size = self
+            .program_arguments
+            .iter()
+            .fold(0, |acc, arg| acc + format!("{}\0", arg).as_bytes().len());
+
+        let rst = memory_ref.write_buffer(address_for_counts, &u32::to_le_bytes(environc));
+        if rst != ErrNo::Success {
+            return rst;
+        }
+        memory_ref.write_buffer(
+            address_for_buffer_size,
+            &u32::to_le_bytes(environ_buf_size as u32),
+        )
+    }
+
+    /// The implementation of the WASI `environ_get` function. It requires an extra `memory_ref` to
+    /// interact with the execution engine.                  
+    pub(crate) fn environ_get(
+        &self,
+        memory_ref: &mut impl MemoryHandler,
+        address_for_string_ptrs: u32,
+        buf_address: u32,
+    ) -> ErrNo {
+        info!("environ_get is called");
+        let buffer = self
+            .environment_variables
+            .iter()
+            .map(|(key, value)| {
+                let environ = format!("{}={}\0", key, value);
+                environ.into_bytes()
+            })
+            .collect::<Vec<_>>();
+        memory_ref.write_string_list(&buffer, buf_address, address_for_string_ptrs)
+    }
+
+    /// THe implementation of the WASI `environ_sizes_get` function. It requires an extra `memory_ref` to
+    /// interact with the execution engine.                  
+    pub(crate) fn environ_sizes_get(
+        &self,
+        memory_ref: &mut impl MemoryHandler,
+        address_for_counts: u32,
+        address_for_buffer_size: u32,
+    ) -> ErrNo {
+        info!("environ_sizes_get is called");
+        let environc = self.environment_variables.len() as u32;
+        let environ_buf_size = self
+            .environment_variables
+            .iter()
+            .fold(0, |acc, (key, value)| {
+                acc + format!("{}={}\0", key, value).as_bytes().len()
+            });
+
+        let rst = memory_ref.write_buffer(address_for_counts, &u32::to_le_bytes(environc));
+        if rst != ErrNo::Success {
+            return rst;
+        }
+        memory_ref.write_buffer(
+            address_for_buffer_size,
+            &u32::to_le_bytes(environ_buf_size as u32),
+        )
+    }
+
+    /// The implementation of the WASI `clock_res_get` function. It requires an extra `memory_ref` to
+    /// interact with the execution engine.                  
+    #[inline]
```

Comment:
I think some of these functions are too big for `inline`, and have grown since originally marked as such.

Reply:
I removed inline for functions with more than 2 lines.